eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeParseMojo.java

```diff
@@ -145,29 +161,18 @@ private XML addFFIs(
      * Add ffi node via xsl transformation and return list of them.
      * @param input Input xmir.
      * @return FFI nodes.
-     * @todo #2649:90min This method may be more general. We need to get rid from rust dependencies
-     *  in this method, because when adding another type of inserts it will be just copy-paste here.
-     *  First of all, the for-loop must create all kinds of FFI nodes, not only {@link RustNode}. I
-     *  think we can implement it, using something like {@code FFINodeFactory}, that will return
-     *  appropriate FFI node for every XML node from {@code nodes}. Also it will be great to move
-     *  paths to XML FFI insert nodes (such as {@code "/program/rusts/rust"}) from this method to
-     *  a static class field.
      * @checkstyle AbbreviationAsWordInNameCheck (8 lines)
      */
     private Collection<FFINode> getFFIs(final XML input) {
-        final List<XML> nodes = this.addFFIs(input).nodes("/program/rusts/rust");
-        final Collection<FFINode> ret = new ArrayList<>(nodes.size());
-        for (final XML node : nodes) {
-            ret.add(
-                new RustNode(
-                    node,
-                    this.names,
-                    this.targetDir.toPath().resolve("Lib"),
-                    this.eoPortalDir.toPath(),
-                    this.generatedDir.toPath().resolve("EOrust").resolve("natives")
-                )
-            );
-        }
+        final Collection<FFINode> ret = new ArrayList<>(0);
+        final XML injected = this.addFFIs(input);
+        this.factory.forEach(
```

Comment:
@levBagryansky let's use just simple `for` here, or `forEach` inside lambda in order to keep the code noncontradictory